docs/reference/contributing/connectivity/NFCEEPROMDriver.md

```diff
@@ -0,0 +1,112 @@
+
+The events `on_bytes_read` and `on_bytes_written` must return the number of bytes successfully read or written in case the maximum size of the operation the device allows is smaller than the request operation. It's up to the caller to call these methods multiple times with appropriate offsets as required.
+
+#### Implementing `read_size` and `write_size`
+
+`read_size` and `write_size` read and write the limits the write and read operations use. Writing or reading beyond the set size must be truncated to the current size.
+
+#### Implementing `erase_bytes`
+
+This is the equivalent of calling `write_bytes` with a buffer filled with `0`. Like `write_bytes`, it must return the number of bytes successfully set to `0`.
+
+### Asynchronous operation
+
+Depending on your hardware, you may support synchronous or asynchronous operation.
+
+We designed the `NFCEEPROMDriver` with asynchronous operation, and the results of long operations are communicated through events. You must use these events, even if your implementation is synchronous.
+
+In asynchronous implementations, you may use an `EventQueue` to schedule processing caused by interrupts. The `NFCTarget` has set up your event queue, which you can get by calling:
+
+`EventQueue* event_queue()`
+
+You may initiate your event processing by calling `call()` on the event queue.
+
+For example:
+
+If your event management occurs in a function called `manage_event`, you should call:
+
+`event_queue()->call(&manage_event);`
+
+The `EventQueue` also accepts passed in parameters and objects for nonstatic method calls. Please see the `EventQueue` documentation for details on calling functions on the queue.
+    
+#### Communicating events back to the NFCTarget
+
+All events call methods in the `Delegate` class object that has been set by the `NFCTarget`. Delegate is accessible through:
+`Delegate* delegate()`
+
+It implements the following methods:
+
+- `void on_session_started(bool success)`
+- `void on_session_ended(bool success)`
+- `void on_bytes_read(size_t count)`
+- `void on_bytes_written(size_t count)`
+- `void on_size_written(bool success)`
+- `void on_size_read(bool success, size_t size)`
+- `void on_bytes_erased(size_t count)`
+
+You must use these to communicate the results of all asynchronous calls.
+
+### Testing
+
+A macro `NFCEEPROM` is required for the test to build. The module containing your driver should contain `mbed_lib.json`. Add a configuration option for the build system to convert into a macro. Your `mbed_lib.json` could look like this:
+
+```javascript
+{
+    "name": "MBED_NFC_<name of your driver>",
+    "config": {
+        "nfceeprom": {
+            "macro_name": "NFCEEPROM",
+            "value": true,
+            "help": "Device supports NFC EEPROM"
+        }
+    }
+}
+```
+
+Run the tests with:
+
+`mbed test -m [mcu] -t [toolchain] -n tests-nfc-eeprom*`.
```

Comment:
Query: Is there any defined or undefined behavior we want to call out in its own section? Also, are there any dependencies we want to call out in their own section?

Reply:
No. All of the HAL target porting guides have defined and undefined behavior sections as part of their template. The connectivity porting guides don't, so this doesn't require them. But if there were something we should call out, I wanted to make sure we did.